VM/vm_params2vm.py

```diff
@@ -0,0 +1,302 @@
+                        "MIN_VS=%s" % (vm_params_dict["min_vs"]),
+                        "TOPO_TYPE=%s\n" % (vm_params_dict["topo_type"]),
+                    ]
+                )
+            )
+        logger.debug("Saved nzvm config file to {}".format(nzvm_cfg_path))
+    else:
+        logger.debug("Path to nzvm.conf should be specified")
+
+
+def main(
+    name, vm_params_path, out_dir, vm_threads=1, logger_name: str = "vm_params2vm"
+):
+    """
+    Orchestrates conversion from vm_params.yaml to VM output
+
+    Parameters
+    ----------
+    name : name of the fault
+    vm_params_path : Path to the vm_params.yaml
+    out_dir : Directory where output files are eventually saved (eg. ..../Data/VMs/Hossack)
+    vm_threads : Number of threads to run NZVM binary with if OpenMP-enabled
+    logger_name : Name of the logger
+
+    Returns
+    -------
+    vm_params_dict : Used for store_summary
+    """
+    logger = qclogging.get_realisation_logger(qclogging.get_logger(logger_name), name)
+
+    # temp directory for current process
+    ptemp = mkdtemp(prefix="_tmp_%s_" % (name), dir=out_dir)
+
+    qclogging.add_general_file_handler(
+        logger, os.path.join(ptemp, "vm_params2vm_{}_log.txt".format(name))
+    )
+
+    (vm_working_dir, nzvm_cfg_path, _) = temp_paths(ptemp)
+
+    with open(vm_params_path, "r") as f:
+        vm_params_dict = yaml.load(f, Loader=yaml.SafeLoader)
+
+    if vm_params_dict is not None:
+        # saves nzvm.cfg
+        save_nzvm_cfg(nzvm_cfg_path, vm_params_dict, vm_working_dir, logger=logger)
+        # makes a copy of vm_params.yaml
+        copyfile(vm_params_path, os.path.join(ptemp, os.path.basename(vm_params_path)))
+        # run the actual generation
+        gen_vm(out_dir, ptemp, vm_threads=vm_threads, logger=logger)
+    else:
+        logger.debug("vm_params.yaml has no data for VM generation")
+
+    vm_params_dict["out_dir"] = out_dir
+
+    # working dir cleanup, return info about VM
+    logger.debug("Cleaning up temp directory {}".format(ptemp))
+    rmtree(ptemp)
+    return vm_params_dict
+
+
+def load_msgs_vm_params(args, logger: Logger = qclogging.get_basic_logger()):
+
+    msgs = []
+    logger.debug("Checking VM params file: {}".format(args.vm_params_path))
+
+    if not os.path.exists(args.vm_params_path):
+        raise FileNotFoundError(args.vm_params_path)
+
+    msgs.append(
+        (
+            args.name,
+            args.vm_params_path,
+            args.out_dir,
+            args.vm_threads,
+            qclogging.get_realisation_logger(logger, args.name).name,
+        )
+    )
+    return msgs
+
+
+def load_args(logger: Logger = qclogging.get_basic_logger()):
+
+    parser = ArgumentParser()
+    arg = parser.add_argument
+
+    arg("name", help="Name of the fault")
+
+    arg(
+        "vm_params_path",
+        help="path to vm_params.yaml",
+    )
+
+    arg("vm_root_dir", help="root directory to place VM files eg. Data/VMs")
```

Comment:
This should be the output directory as this script should only be operating on a single vm

Reply:
This can also default to the folder the vm_params file is in, if none is given